atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/DbAtlasDbFactory.java

```diff
@@ -96,6 +98,15 @@ public TimestampService createTimestampService(
         return PersistentTimestampServiceImpl.create(createTimestampBoundStore(timestampTable, dbkvs));
     }
 
+    @Override
+    public TimestampManagementService createTimestampManagementService(TimestampService timestampService) {
+        Preconditions.checkArgument(timestampService instanceof PersistentTimestampService,
+                "TimestampManagementService must be created based on result of createTimestampService call."
+                        + "\nExpected a PersistentTimestampServiceImpl, got %s", timestampService.getClass());
```

Comment:
nit: as above, probably should update the preconditions message; also not sure if we should just check if it's an instance of TMS as opposed to PTS specifically